logprep/util/getter.py

```diff
@@ -154,15 +168,32 @@ def get_raw(self) -> bytearray:
         resp = None
         while resp is None:
             try:
-                resp = session.get(
-                    url=url,
-                    timeout=5,
-                    allow_redirects=True,
-                    headers=self._headers,
-                )
+                if not self._found_valid_token and self._tokens:
+                    resp = self._search_for_valid_token(session, url)
+                else:
+                    resp = self._execute_request(session, url)
```

Comment:
this confuses me. `_search_for_valid_token` obvisously does more than searching for a valid token. please refactor for cleaner code and better understanding

Reply:
I revised it a bit and added some comments. Please have a look at it again.